### PR TITLE
Add support for /proc/[pid]/limits

### PR DESCRIPTION
**Pull request by jseba**
This adds support for the `/proc/[pid]/limits` file to view information from `ulimit` or `setrlimit`. It prints the output in the exact format that the Linux kernel uses, which is hand-spaced columns instead of tabs or a columnar printer. In order to support the file format, helpers have been added to get the textual names and units for the limit types.

### CODE REVIEW

**Comment by avagin** [comment]
Need to add a test case in https://github.com/google/gvisor/blob/master/test/syscalls/linux/rlimits.cc.

File: pkg/sentry/fsimpl/proc/task_files.go [hunk 1]
```diff
@@ -529,6 +529,43 @@ func (fd *memFD) SetStat(context.Context, vfs.SetStatOptions) error {
...
+	task *kernel.Task
+}
+
+func (d *limitsData) Generate(ctx context.Context, buf *bytes.Buffer) error {
+	// formatting matches the kernel output from linux/fs/proc/base.c:proc_pid_limits()
+	fmt.Fprintf(buf, "Limit                     Soft Limit           Hard Limit           Units     \n")
+	for _, lt := range limits.AllLimitTypes {
+		fmt.Fprintf(buf, "%-25s ", lt.Name())
+
+		l := d.task.Limits().Get(lt)
```
Comment (avagin) [inline pkg/sentry/fsimpl/proc/task_files.go:547]: d.taks.Limits() can be called once before the loop.

**Comment by avagin** [comment]
The PR looks good overall.

**Comment by jseba** [comment]
> Need to add a test case in https://github.com/google/gvisor/blob/master/test/syscalls/linux/rlimits.cc.

Can do, forgot about the C++ tests. What test case were you thinking of? That changing the various limits shows up in the proc file?

**Comment by avagin** [comment]
@jseba I think about something very basic: read a file and check one or two limits are shown properly.

**Comment by jseba** [comment]
Apologies for taking so long to get back to this, I rebased and added a simple unit test to read the limits file after changing it via setrlimit

**Comment by jseba** [comment]
@avagin friendly ping when you get a chance to review again, I'd appreciate it

File: test/util/proc_util.cc [hunk 2]
```diff
@@ -331,5 +331,159 @@ bool IsTHPDisabled() {
   return StackTHPDisabled(maps.ValueOrDie());
 }
 
+PosixErrorOr<ProcLimitsEntry> ParseProcLimitsLine(absl::string_view line) {
+    ProcLimitsEntry limits_entry = {};
+
+    auto const nameColumnWidth = 25;
```
Comment (avagin) [inline test/util/proc_util.cc:337]: unused variable 'nameColumnWidth' [-Werror,-Wunused-variable]
  auto const nameColumnWidth = 25;
             ^

**Comment by avagin** [comment]
@jseba pls merge the fix to the proper commit.

**Comment by jseba** [comment]
> @jseba pls merge the fix to the proper commit.

Do I rebase it into the unit test commit and force push?

**Comment by avagin** [comment]
@jseba yes. We want to have the clean history in the main repo.

**Comment by jseba** [comment]
> @jseba yes. We want to have the clean history in the main repo.

gotcha, not sure how copybara works but I squashed the unit test fix into its commit, let me know if I need to squash the whole PR into one commit

File: pkg/sentry/limits/limits.go [hunk 3]
```diff
@@ -43,6 +43,103 @@ const (
 	Rttime
 )
 
+var AllLimitTypes = []LimitType{
```
Comment (avagin) [inline pkg/sentry/limits/limits.go:46]: exported var AllLimitTypes should have comment or be unexported

File: test/syscalls/linux/BUILD [hunk 4]
```diff
@@ -2139,6 +2139,7 @@ cc_binary(
     linkstatic = 1,
     deps = [
         "//test/util:capability_util",
+        "//test/util:proc_util",
```
Comment (avagin) [inline test/syscalls/linux/BUILD:2142]: need to add "@com_google_absl//absl/strings" too

**Pull request merged**